python/triton/tools/aot.py

```diff
@@ -83,24 +83,31 @@
         print(module)
         sys.exit(0)
 
-    if not args.sm:
-        raise argparse.ArgumentError(None, "Must specify --sm for PTX compilation")
+    # set arch depending on platform
+    if args.gfx:
+        arch = args.gfx
+    elif args.sm:
+        arch = args.sm
+    else:
+        raise argparse.ArgumentError(None, "Must specify --sm or --gfx for ttgir compilation")
```

Comment:
Consider refactor code above too: https://github.com/ROCmSoftwarePlatform/triton/pull/210/files#diff-3df9553c27ea5bf270c80c9193bb384cbe2561a669ac04a65918124bd83aff6fR49-R84

It masks AMD GPU cases, so this changes is unused.

Reply:
I am not sure I understand. I added this as a fix for this[ lit test](https://github.com/ROCmSoftwarePlatform/triton/pull/210/files#diff-8d11e23ee6ecc001d8181f4b28b20b569489e31779837e068b7709b8c5a7d85dR1). The issue is related to the path taken by when the target is llvmir. The AMDGCN path exists [before ](https://github.com/ROCmSoftwarePlatform/triton/blob/275fead8e3dee1d9e0b7201cfcccc5de24685466/python/triton/tools/aot.py#L84 )this code is hit. Can you add more detail?